devito/operator/profiling.py

```diff
@@ -367,21 +367,23 @@ class AdvisorProfiler(AdvancedProfiler):
     _ext_calls = [_api_resume, _api_pause]
 
     def __init__(self, name):
-        self.path = locate_intel_advisor()
-        if self.path is None:
-            self.initialized = False
+        if self._attempted_init:
+            return
+
+        super().__init__(name)
+
+        path = get_advisor_path()
+        if path:
+            self._include_dirs.append(path.joinpath('include').as_posix())
+            self._lib_dirs.append(path.joinpath('lib64').as_posix())
+            self._attempted_init = True
```

Comment:
maybe for another day, but these three lines 

```
     self._attempted_init = True
else:
     self._attempted_init = False
```

could become a single line

```
self._attempted_init = bool(path)
```

but anyway, nitpicking... maybe you can do it in another PR

Reply:
but why append then if path is None? If the compiler reverts to advanced in the case of advisor's absence, the paths will be linked right?